importlib_metadata/__init__.py

```diff
@@ -475,9 +475,18 @@ def make_file(name, hash=None, size_str=None):
 
         @pass_none
         def make_files(lines):
-            return list(starmap(make_file, csv.reader(lines)))
-
-        return make_files(self._read_files_distinfo() or self._read_files_egginfo())
+            return list(
+                filter(
+                    lambda package_path: package_path.locate().exists(),
+                    list(starmap(make_file, csv.reader(lines))),
```

Comment:
nit: This `list()` is no longer necessary.

Reply:
Will remove.

Reply:
Removed in https://github.com/python/importlib_metadata/pull/437/commits/61eca31a7456ccc39e3fda8e098497a4f28482b4